src/domain/synthetics/trade/useHighExecutionFeeAcknowledgement.tsx

```diff
@@ -0,0 +1,31 @@
+import { Trans } from "@lingui/macro";
+import Checkbox from "components/Checkbox/Checkbox";
+import { getVeryHighExecutionFee } from "config/chains";
+import { BigNumber } from "ethers";
+import { useChainId } from "lib/chains";
+import { USD_DECIMALS } from "lib/legacy";
+import { expandDecimals } from "lib/numbers";
+import { useMemo, useState } from "react";
+
+export function useHighExecutionFeeAcknowledgement(executionFeeUsd: BigNumber | undefined) {
+  const [isHighExecutionFeeAccepted, setIsHighExecutionFeeAccepted] = useState(false);
+  const { chainId } = useChainId();
+  const veryHighExecutionFeeUsd = useMemo(
+    () => expandDecimals(getVeryHighExecutionFee(chainId), USD_DECIMALS),
+    [chainId]
+  );
+  const shouldAccept = executionFeeUsd?.gte(veryHighExecutionFeeUsd);
+
+  return {
+    highExecutionFeeNotAcceptedError: shouldAccept && !isHighExecutionFeeAccepted,
+    element: shouldAccept ? (
+      <div>
+        <Checkbox asRow isChecked={isHighExecutionFeeAccepted} setIsChecked={setIsHighExecutionFeeAccepted}>
+          <span className="text-warning font-sm">
+            <Trans>Acknowledge very high network Fees</Trans>
+          </span>
+        </Checkbox>
+      </div>
+    ) : null,
+  };
+}
```

Comment:
Few suggestions (only if you think it will improve the readability of the hook)

- Using the name `useHighExecutionFeeConsent` (a bit smaller and conveys the exact same meaning)
- The name`highExecutionFeeNotAcceptedError` signifies it will return some text but it will return boolean so maybe call it something like `isHighFeeConsentPending` or `isConsentRequiredAndPending` (if I understand it correctly)

Reply:
1. fixed
2. changed to isHighFeeConsentError